realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncConfigurationTests.kt

```diff
@@ -424,4 +431,57 @@ class SyncConfigurationTests {
                 .build()
         assertTrue(configuration.isAllowWritesOnUiThread)
     }
+
+    @Test
+    fun coroutinesFactory_defaultNonNull() {
+        val configuration = SyncConfiguration.Builder(createTestUser(app), DEFAULT_PARTITION)
+                .build()
+        assertNotNull(configuration.flowFactory)
+    }
+
+    @Test
+    fun coroutinesFactory() {
```

Comment:
Same, missing test for what happens if you assign null

Reply:
This is a Kotlin class, so non-nullability is enforced after adding `@Nonnull` to the Java method.

Reply:
No, it doesn't prevent that when calling it from Java. At best you get a warning which can be missed or ignored.